thirdeye/thirdeye-frontend/app/pods/manage/alert/edit/controller.js

```diff
@@ -330,27 +314,29 @@ export default Controller.extend({
    * Actions for edit alert form view
    */
   actions: {
+
     /**
      * Make sure alert name does not already exist in the system
      * @method validateAlertName
      * @param {String} name - The new alert name
+     * @param {Boolean} userModified - Up to this moment, is the new name auto-generated, or user modified?
+     * If user-modified, we will stop modifying it dynamically (via 'isAlertNameUserModified')
      * @return {undefined}
      */
     validateAlertName(name) {
-      const originalName = this.get('alertFunctionName');
       let isDuplicateName = false;
-      if (name === originalName) { return; }
-
       this.fetchAlertByName(name).then(alert => {
         for (var resultObj of alert) {
           if (resultObj.functionName === name) {
             isDuplicateName = true;
           }
         }
+        // Either add or clear the "is duplicate name" banner
         this.set('isAlertNameDuplicate', isDuplicateName);
```

Comment:
I know this is not part of the PR, but is there a way to short-circuit this? It seems like you only need to find a duplicate and then set `isAlertNameDuplicate` to true immediately without needing to iterate through the rest of the `alert` object.

Reply:
Hey, good point. I've removed the loop in favor of a find.